vault.go

```diff
@@ -30,54 +104,51 @@ path "{{ $instance }}/keys/{{ $profile }}" {
 }
 {{- end }}
 `
-	t, _ := template.New("policy").Parse(policy)
+
+func generatePolicy(name string, minioInstances []string) (string, error) {
+
+	t, _ := template.New("policy").Parse(POLICY_TEMPLATE)
 	w := bytes.NewBufferString("")
 
-	t.Execute(w, struct {
-		MountName      string
+	err := t.Execute(w, struct {
 		ProfileName    string
 		MinioInstances []string
-	}{MountName: mountName, ProfileName: name, MinioInstances: minioInstances})
+	}{ProfileName: name, MinioInstances: minioInstances})
 
-	return fmt.Sprintf("%s", name), w.String()
-}
+	if err != nil {
+		return "", err
+	}
 
-func cleanName(name string) string {
-	return strings.ReplaceAll(name, "_", "-")
+	return w.String(), nil
 }
 
-func doPolicy(vc *vault.Client, name, mountName string, minioInstances []string) (string, error) {
-	policyName, policy := generatePolicy(name, mountName, minioInstances)
+// Writes a policy to Vault
+func (vc *VaultConfigurerStruct) doPolicy(name string, mountName string) (string, error) {
+	policy, err := generatePolicy(name, vc.MinioInstances)
```

Comment:
Does this return an error if the policy doesn't exist? Just want to make sure it won't ever not create policies.

Reply:
the generatePolicy()? I added error handling if the templating through an error.